ql/src/test/org/apache/hadoop/hive/ql/TestTxnCommands2.java

```diff
@@ -3080,6 +3084,83 @@ public void testAcidOrcWritePreservesFieldNames() throws Exception {
     hiveConf.setBoolVar(HiveConf.ConfVars.HIVE_VECTORIZATION_ENABLED, true);
   }
 
+  @Test
+  public void testFailureScenariosCleanupCTAS() throws Exception {
+    boolean[] booleans = {true, false};
+    for (boolean var1 : booleans) {
```

Comment:
do we really need to cover all 16 combinations?

Reply:
Its always better to check all cases.

Reply:
Done. Nice use of bitwise operators :) .